### DISCUSSION TITLE
Setting distinct border colours for scratchpads

### DISCUSSION THREAD

**Discussion by amazingwilf** [bakkeby/dwm-flexipatch#246] [Ideas]
Hi,

Love this project - it's made DWM useable for me! 

I suffer from a visual impairment, which makes it difficult for me to distinguish between different clients, so I define different coloured borders for different client types. I've worked out how to do this by adding an additional scheme for sticky windows, but I can't work out how to do this for scratchpads. Ideally I'd like to be able to use it similarly to your renamedscratchpads patch, with schemes for normal and selected scratchpads (in fact, what I've been doing so far is to leave scratchpads out of the flexipatch build, and add renamedscratchpads afterwards, but this doesn't play nice with dwmc and sxhkd, which is how I generally manage keyboard shortcuts).

How feasible would it be to add this functionality to the scratchpads patch in flexipatch? Or could you give me a pointer as to how to apply it myself?

Much obliged, and thanks so much for the project!

David

**Comment by bakkeby** [comment]
I suppose that it would be feasible to include the renamedscratchpads patch into dwm-flexipatch - it would likely just be a bit more involved due to compatibility with other patches. Just adding the colour scheme is also an option.

I would invite you to try out [dusk](https://github.com/bakkeby/dusk) which already have all of this built-in.

> but this doesn't play nice with dwmc and sxhkd

Granted you are going to have problems like this with the (re-)named scratchpads patch as it depends on having a predefined default application that is to run if a scratchpad window does not already exists.

A workaround for this could be to have wrapper functions like this that you would refer to in the fsignals list or in the list of IPC commands depending on what you use:

```c
static void
togglescratch_w(const Arg *arg)
{
	togglescratch(&((Arg) { .v = spcmd_w }));
}
```

(this function you could have in your config.h for example)

The scratchpads patch (that comes with dwm-flexipatch) does not have this issue as the commands are arranged in an array.

**Reply by amazingwilf** [reply]
Thanks for the reply - I'll give the wrapper workaround a try!

**Reply by amazingwilf** [reply]
Works a treat! Thank you, I have achieved personal DWM nirvana!

I am going to give dusk a try, anyway, though.
